### ISSUE TITLE
lots of "TODO: use x from MacCompileStuff" comments

### ISSUE THREAD

**Issue by cooljeanius** [hrubymar10/MacCompileStuff#1]
Currently `SourcesAndScripts/Scripts/functions` contains a lot of checks for things being installed in `/usr/local`, and suggests to use Homebrew to install things that are missing. However, users might not use Homebrew as their package manager (I use MacPorts for instance, and then there's also Fink), and even if they do use Homebrew, it could be installed to a non-default prefix. And in any case, MacCompileStuff provides all the packages being requested itself, so it should just use its own packages when possible (as the TODO comments say) instead of depending on an external package manager.

**Comment by hrubymar10** [comment]
Yes, that applies to `master` branch, but it should be solved in `universal_libs_support` branch which is default for `1.15` and future `1.16` https://github.com/hrubymar10/MacCompileStuff/blob/universal_libs_support/SourcesAndScripts/Scripts/env

Is that solution to this issue?

**Comment by cooljeanius** [comment]
ok switching to the `universal_libs_support` branch gets me farther, but then leads to a build failure when freetype attempts to link against non-universal libs installed in /opt/local by MacPorts:

```
libtool: link: clang -dynamiclib  -o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/libfreetype.6.dylib  /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftsystem.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftdebug.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftinit.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftbase.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftbbox.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftbdf.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftbitmap.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftcid.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftfstype.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftgasp.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftglyph.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftgxval.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftmm.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftotval.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftpatent.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftpfr.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftstroke.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftsynth.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/fttype1.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftwinfnt.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/truetype.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/type1.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/cff.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/type1cid.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/pfr.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/type42.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/winfnt.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/pcf.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/bdf.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/sfnt.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/autofit.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/pshinter.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/raster.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/smooth.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftcache.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftgzip.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftlzw.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/ftbzip2.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/psaux.o /Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/psnames.o   -L/Users/ericgallager/Documents/GitHub/MacCompileStuff/lib -L/opt/local/lib -lz -lbz2 -lpng  -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -mmacosx-version-min=10.11   -install_name  /usr/local/lib/libfreetype.6.dylib -compatibility_version 24 -current_version 24.4 -Wl,-single_module -Wl,-exported_symbols_list,/Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/.libs/libfreetype-symbols.expsym
ld: warning: ignoring file /opt/local/lib/libz.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /opt/local/lib/libbz2.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: in '/opt/local/lib/libunwind.dylib', building for macOS-arm64 but attempting to link with file built for macOS-x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/ericgallager/Documents/GitHub/MacCompileStuff/SourcesAndScripts/Sources/freetype-2.10.4-arm64/objs/libfreetype.la] Error 1
```

I'll have to see if I can edit the script to get it to ignore my MacPorts prefix...

**Comment by hrubymar10** [comment]
`universal_libs_support` can be build only on native Apple Silicon mac. It was necessary at time of creating that branch, because some libraries did not fully support cross-compiling arm targets yet. I expect that situation is better now, but I don't think that everything is solved. The most problems IIRC was with GNOME libraries built using meson+ninja.

**Comment by cooljeanius** [comment]
> `universal_libs_support` can be build only on native Apple Silicon mac. It was necessary at time of creating that branch, because some libraries did not fully support cross-compiling arm targets yet. I expect that situation is better now, but I don't think that everything is solved.

Could you check up on said situation again please, @hrubymar10?

**Comment by hrubymar10** [comment]
I will close this issue in favor of #3 . There is updated info regarding cross compilation.

TL;DR: I won't invest time in it. I don't have enough time for it and also I don't find it important.
